distributed/deploy/local.py

```diff
@@ -280,9 +285,10 @@ def _start_worker(self, death_timeout=60, **kwargs):
 
         if self.processes:
             W = Nanny
+            kwargs["worker_class"] = self.worker_class
             kwargs["quiet"] = True
         else:
-            W = Worker
+            W = self.worker_class or Worker
```

Comment:
If you accept the other changes, then we can probably drop the conditional above (it's effectively now handled in the constructor)

We might also remove the `self.processes` attribute, which seems to be effectively replaced and generalized by `self.worker_class`.

We should have something like the following though to maintain previous behavior.

```python
if self.worker_class is Nanny:
    kwargs['quiet'] = True
```

Reply:
Sure, your changes look fine to me. I will push a commit.